controller/user.js

```diff
@@ -172,28 +172,24 @@ async function verifyAccount(req, res) {
 }
 
 async function update(req, res, next) {
-  const id = getIdParam(req);
-  if (!req.body.id) {
-    throw new errors.FcError(errors.MISSING_USER_ID);
-  } else {
-    models.User.update(req.body, {
-      where: { id: id },
-    })
-      .then((rowsUpdate) => {
-        if (rowsUpdate[0] == 0) {
-          throw new errors.FcError(errors.USER_NOT_FOUND);
-        } else {
-          models.User.findByPk(id).then((user) => {
-            res.status(statusCode.SUCCESS).json({
-              code: 0,
-              msg: "",
-              data: { user: buildUserJson(user) },
-            });
-          });
-        }
-      })
-      .catch(next);
+  const user_id = req.user.id;
+  const user = await models.User.findByPk(user_id);
+  if (!user) {
+    throw new errors.FcError(errors.USER_NOT_FOUND);
   }
+
+  if (req.body.full_name !== undefined) user.full_name = req.body.full_name;
+  if (req.body.school_name !== undefined) user.school_name = req.body.school_name;
+  user.save().then(updatedUser => {
+    if (!updatedUser) {
+      throw new errors.FcError(errors.SYSTEM_ERROR);
+    }
```

Comment:
Remove this unnecessary check.

If error occurs, handle it in `catch` section.